src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -1099,15 +1098,14 @@ static bool isOrContainsErrorType(TypeSymbol typeSymbol)
         /// </summary>
         private bool IsValidOverrideReturnType(Symbol overridingSymbol, TypeWithAnnotations overridingReturnType, TypeWithAnnotations overriddenReturnType, DiagnosticBag diagnostics)
         {
-            if (DeclaringCompilation.LanguageVersion >= MessageID.IDS_FeatureCovariantReturnsForOverrides.RequiredVersion())
+            if (overridingSymbol.ContainingAssembly.RuntimeSupportsCovariantReturnsOfClasses &&
+                DeclaringCompilation.LanguageVersion >= MessageID.IDS_FeatureCovariantReturnsForOverrides.RequiredVersion())
```

Comment:
Can we use the `CheckFeatureAvailability` helper instead? Allowing indirection here as helped us produce better diagnostics in the past.

Reply:
No, because the purpose of this code is not to report a language version diagnostic.

---
In reply to: [437696307](https://github.com/dotnet/roslyn/pull/44025#discussion_r437696307) [](ancestors = 437696307)